lib/1.4/dml-builtins.dml

```diff
@@ -1858,6 +1861,50 @@ template bank is (object, shown_desc, _qname) {
     shared method transaction_access(transaction_t *t, uint64 offset,
                                      void *aux) -> (exception_type_t) default {
         local uint64 size = SIM_transaction_size(t);
+
+        if (size <= 8) {
+            return _transaction_access(t, offset, aux) ?
+                Sim_PE_No_Exception : Sim_PE_IO_Not_Taken;
+        }
+
+        local uint8 buf[size];
+        if (SIM_transaction_is_write(t)) {
+            local buffer_t bytes;
+            bytes.data = buf;
+            bytes.len = size;
+            SIM_get_transaction_bytes(t, bytes);
+        }
+
+        local int sz = 8 - (offset % 8);
+        local uint64 offs = 0;
+        while (offs < size) {
+            local atom_t atoms[3] = {
+                ATOM_data(&buf[offs]),
```

Comment:
`_transaction_access` only uses size, len and data; easier to extract them here and pass as args instead of building a transaction around them. At this stage, any extensions of the transaction are in the aux arg anyway.

Reply:
Ok.

Reply:
Actually, `_transaction_access` gets and sets the value of the transaction too. By building a smaller transaction I can wave my hand at those things and just use the transaction api-calls.

Reply:
I gave it a try myself and pushed a patch that uses a buffer.

Reply:
Thanks, looks good!
Funny thing: I can't approve your changes, but you can - is this a sneaky way of bypassing the review-requirements?